model/src/main/kotlin/ScanResult.kt

```diff
@@ -49,4 +55,53 @@ data class ScanResult(
     @JsonAlias("rawResult")
     @JsonInclude(JsonInclude.Include.NON_NULL)
     val rawResult: JsonNode? = null
-)
+) {
+    /**
+     * Filter all detected licenses and copyrights from the [summary] which are below [path], and set the [path] for
+     * [provenance]. Findings in files which are listed in [LICENSE_FILE_NAMES] are also kept.
+     */
+    fun filterPath(path: String): ScanResult {
+        val matchersForLicenseFiles = LICENSE_FILE_NAMES.map {
+            FileSystems.getDefault().getPathMatcher("glob:$it")
+        }
+
+        val pathToMatch = Paths.get(path)
+
+        fun SortedSet<TextLocation>.filterPath() =
```

Comment:
I prefer to keep the helper function here as it is specific to this function. However, with the upcoming change of scanning by repository it might be refactored anyway.

Reply:
not 100% convinced but this is a matter of personal preference a we have no guideline for the code style defined.